File: Benchmarking/src/gen543.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: [s0]
    var p1: [[s0]]
  }
  struct s3 {
    var p0: [s0]
    var p1: [[s1]]
  }
  func f16(_ v0: s3) -> s3 {
    var v3: s3 = v0
    var v1: s3 = v0
    var v8: s3 = v0
    var v6: s3 = v1
    var v7: s3 = v3
    let v5: [s0] = v0.p0
    v1.p0 = v5
    v7.p0 = v5
    var v2: s3 = v1
    v1.p0 = v5
    v1 = v2
    v7.p0 = v5
    v1.p0 = v5
    let v20: [s3] = [v7, v1, v2, v2, v2, v8, v6]
    var v17: [s0] = v5
    v17 = v5
    v8.p0 = v17
    let v33: [[s1]] = v3.p1
    let v67: s3 = v20[3]
    v7.p1 = v33
    return v67
  }
  @inline(never)
  func f0(_ v0: [s3], _ v1: Double) -> Double {
    let v5: s3 = v0[1]
    var v4: Double = v1
    let v8: s3 = f16(v5)
    let v20: [s0] = v8.p0
    let v11: s0 = v20[2]
    let v41: [[Double]] = v11.p1
    let v14: [Double] = v41[2]
    let v51: Double = v14[0]
    let v195: Double = v51 / v4
    return v195
  }
  func benchmark() {
    let v0: [s3] = [s3(p0: [s0(p0: [[0.0]], p1: [[1.0], [2.0], [3.0]]), s0(p0: [[4.0]], p1: [[5.0], [6.0], [7.0]]), s0(p0: [[8.0]], p1: [[9.0], [10.0], [11.0]])], p1: [[s1(p0: [s0(p0: [[12.0]], p1: [[13.0], [14.0], [15.0]]), s0(p0: [[16.0]], p1: [[17.0], [18.0], [19.0]]), s0(p0: [[20.0]], p1: [[21.0], [22.0], [23.0]])], p1: [[s0(p0: [[24.0]], p1: [[25.0], [26.0], [27.0]])]])]]), s3(p0: [s0(p0: [[28.0]], p1: [[29.0], [30.0], [31.0]]), s0(p0: [[32.0]], p1: [[33.0], [34.0], [35.0]]), s0(p0: [[36.0]], p1: [[37.0], [38.0], [39.0]])], p1: [[s1(p0: [s0(p0: [[40.0]], p1: [[41.0], [42.0], [43.0]]), s0(p0: [[44.0]], p1: [[45.0], [46.0], [47.0]]), s0(p0: [[48.0]], p1: [[49.0], [50.0], [51.0]])], p1: [[s0(p0: [[52.0]], p1: [[53.0], [54.0], [55.0]])]])]])]
    let v1: Double = 56.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 56.0
    for _ in 1...1000 {
      result = f0(v0, v1)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()

File: Benchmarking/src/gen543.scala
import java.lang.System.nanoTime
import scala.collection.immutable.Vector
object Gen extends App {
  case class s0 (
    p0: Vector[Vector[Double]],
    p1: Vector[Vector[Double]]
  )
  case class s1 (
    p0: Vector[s0],
    p1: Vector[Vector[s0]]
  )
  case class s3 (
    p0: Vector[s0],
    p1: Vector[Vector[s1]]
  )
  def f16(v0: s3): s3 = {
    var v3: s3 = v0
    var v1: s3 = v0
    var v8: s3 = v0
    var v6: s3 = v1
    var v7: s3 = v3
    val v5: Vector[s0] = v0.p0
    v1 = v1.copy(p0 = v5)
    v7 = v7.copy(p0 = v5)
    var v2: s3 = v1
    v1 = v1.copy(p0 = v5)
    v1 = v2
    v7 = v7.copy(p0 = v5)
    v1 = v1.copy(p0 = v5)
    val v20: Vector[s3] = Vector(v7, v1, v2, v2, v2, v8, v6)
    var v17: Vector[s0] = v5
    v17 = v5
    v8 = v8.copy(p0 = v17)
    val v33: Vector[Vector[s1]] = v3.p1
    val v67: s3 = v20(3)
    v7 = v7.copy(p1 = v33)
    v67
  }
  @noinline
  def f0(v0: Vector[s3], v1: Double): Double = {
    val v5: s3 = v0(1)
    var v4: Double = v1
    val v8: s3 = f16(v5)
    val v20: Vector[s0] = v8.p0
    val v11: s0 = v20(2)
    val v41: Vector[Vector[Double]] = v11.p1
    val v14: Vector[Double] = v41(2)
    val v51: Double = v14(0)
    val v195: Double = v51 / v4
    v195
  }
  def benchmark(): Unit = {
    val v0: Vector[s3] = Vector(s3(Vector(s0(Vector(Vector(0.0)), Vector(Vector(1.0), Vector(2.0), Vector(3.0))), s0(Vector(Vector(4.0)), Vector(Vector(5.0), Vector(6.0), Vector(7.0))), s0(Vector(Vector(8.0)), Vector(Vector(9.0), Vector(10.0), Vector(11.0)))), Vector(Vector(s1(Vector(s0(Vector(Vector(12.0)), Vector(Vector(13.0), Vector(14.0), Vector(15.0))), s0(Vector(Vector(16.0)), Vector(Vector(17.0), Vector(18.0), Vector(19.0))), s0(Vector(Vector(20.0)), Vector(Vector(21.0), Vector(22.0), Vector(23.0)))), Vector(Vector(s0(Vector(Vector(24.0)), Vector(Vector(25.0), Vector(26.0), Vector(27.0))))))))), s3(Vector(s0(Vector(Vector(28.0)), Vector(Vector(29.0), Vector(30.0), Vector(31.0))), s0(Vector(Vector(32.0)), Vector(Vector(33.0), Vector(34.0), Vector(35.0))), s0(Vector(Vector(36.0)), Vector(Vector(37.0), Vector(38.0), Vector(39.0)))), Vector(Vector(s1(Vector(s0(Vector(Vector(40.0)), Vector(Vector(41.0), Vector(42.0), Vector(43.0))), s0(Vector(Vector(44.0)), Vector(Vector(45.0), Vector(46.0), Vector(47.0))), s0(Vector(Vector(48.0)), Vector(Vector(49.0), Vector(50.0), Vector(51.0)))), Vector(Vector(s0(Vector(Vector(52.0)), Vector(Vector(53.0), Vector(54.0), Vector(55.0))))))))))
    val v1: Double = 56.0
    val start = nanoTime()
    var result: Double = 56.0
    (1 to 1000).foreach { _ =>
      result = f0(v0, v1)
    }
    val end = nanoTime()
    println(result)
    println(end - start)
  }
  benchmark()
}